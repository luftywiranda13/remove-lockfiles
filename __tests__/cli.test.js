'use strict';

const execa = require('execa');
const fs = require('fs-extra');
const tempy = require('tempy');

describe('CLI', () => {
  describe('no lockfile', () => {
    test('output', async () => {
      const stdout = await execa.stdout('./cli.js');

      expect(stdout).toMatch(/No lockfile found/);
    });
  });

  describe('with lockfiles', () => {
    test('output', async () => {
      const tempDir = tempy.directory();
      await fs.copy(`${__dirname}/fixtures`, tempDir);
      const stdout = await execa.stdout('./cli.js', [tempDir]);

      expect(stdout).not.toMatch(/npm-shrinkwrap.json/);
    });

    test('shrinkwrap option', async () => {
      const tempDir = tempy.directory();
      await fs.copy(`${__dirname}/fixtures`, tempDir);
      const stdout = await execa.stdout('./cli.js', [tempDir, '--shrinkwrap']);

      expect(stdout).toMatch(/npm-shrinkwrap.json/);
    });
  });
});
