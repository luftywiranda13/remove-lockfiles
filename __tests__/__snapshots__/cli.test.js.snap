// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI \`--shrinkwrap\` flag 1`] = `
"✔ Removed:
npm-shrinkwrap.json"
`;

exports[`CLI lockfiles found 1`] = `
"✔ Removed:
package-lock.json
yarn.lock"
`;

exports[`CLI no lockfile 1`] = `"ℹ No lockfile found"`;
